llvm/include/llvm/Analysis/InlineOrder.h

```diff
@@ -32,7 +32,86 @@ template <typename T> class InlineOrder {
 };
 
 std::unique_ptr<InlineOrder<std::pair<CallBase *, int>>>
-getInlineOrder(FunctionAnalysisManager &FAM, const InlineParams &Params);
+getInlineOrder(FunctionAnalysisManager &FAM, const InlineParams &Params,
+               ModuleAnalysisManager &MAM, Module &M);
+
+/// Used for dynamically registering InlineOrder as plugins
+///
+/// An inline order plugin adds a new inline order at runtime by registering
+/// an instance of PluginInlineOrderAnalysis in the current
+/// ModuleAnalysisManager. For example, the following code dynamically registers
+/// the default inline order:
```

Comment:
> Sorry, I am slow to understand this. Is the following correct understanding?
> 
> PluginInlineOrderAnalysis is always there, regardless of whether you have a plugin. If you actually have a plugin, we can use Factory in PluginInlineOrderAnalysis to construct a custom inline order.
> 
> Now, why are we using an analysis here? Is that because it's a good hook to use?

Reply:
We should explain here that we need an analysis so that we can use the PassManager's ability to load plugins.